src/components/Calculator.js

```diff
@@ -1,30 +1,45 @@
+import { useState } from 'react';
+import calculate from '../logic/calculate';
 import './Calculator.css';
+// import calculate from '../logic/calculate';
+// import operate from '../logic/operate';
 
-const Calculator = () => (
-  <div>
-    <div className="grid-container">
-      <div className="grid-input">0</div>
-      <div className="grid-item grid-2">AC</div>
-      <div className="grid-item grid-3">+/-</div>
-      <div className="grid-item grid-4">%</div>
-      <div className="div-item right">/</div>
-      <div className="grid-item grid-6">7</div>
-      <div className="grid-item grid-7">8</div>
-      <div className="grid-item grid-8">9</div>
-      <div className="mul-item right">X</div>
-      <div className="grid-item grid-10">4</div>
-      <div className="grid-item grid-11">5</div>
-      <div className="grid-item grid-12">6</div>
-      <div className="sub-item right">-</div>
-      <div className="grid-item grid-14">1</div>
-      <div className="grid-item grid-15">2</div>
-      <div className="grid-item grid-16">3</div>
-      <div className="add-item right">+</div>
-      <div className="grid-item grid-0">0</div>
-      <div className="grid-item grid-19">.</div>
-      <div className="equ-item right">=</div>
+function Calculator() {
+  const [state, setState] = useState({ total: 0, next: null, operation: null });
+  const clickHandler = (e) => {
+    setState((state) => calculate(state, e.target.name));
+  };
+
+  return (
+    <div>
+      <div className="grid-container">
+        <div className="result">
+          {state.total}
+          {state.operation}
+          {state.next}
+        </div>
```

Comment:
When the AC button is clicked the name property (string 'ac') is displayed and any further calculation produces an error.
See the image below.

To reproduce the error, enter one of the operators after the last digit on the screen:

![image](https://user-images.githubusercontent.com/20554583/219379158-44f211f3-59bc-4f76-8f48-b2a1df633ffb.png)

![image](https://user-images.githubusercontent.com/20554583/219379167-f40897bb-14a4-4fae-b0f0-fed909c3d286.png)